### DISCUSSION TITLE
Is it possible to add @click.prevent conditionally?

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#3836] [1. Help]
In my situation, the `<template>` loop goes over different item types. In one case, an additional property `@click.prevent` should be added to the template code. Is that possible without having to create a new template (I want to reuse it as much as possible)?

**Comment by simonferndriger** [comment]
I would just use `@click` and add the `event.preventDefault()` conditionally - or is there a better (more AlpineJS) way?

**Reply by ekwoka** [reply]
That would be the way.

or do an x-bind and conditionally assign the key.

**Reply by simonferndriger** [reply]
Thanks. How would the later look like (conditionally assigning the key I mean)? Because as far as I know I either define a key or don't. Not sure how to add it later...

![image](https://github.com/alpinejs/alpine/assets/1435963/4abad0cb-f52f-4da2-80d0-4dc28289bb01)

**Comment by james0r** [comment]
Might not be helpful, but sometimes in PHP I conditionally render attributes with a little helper

```php
function attrsEncode($attrs) {
    $html = '';

    foreach ($attrs as $key => $value) {
      $html .= sprintf('%s="%s" ', $key, $value);
    }

    return $html;
  }
  ```
  
  Usage:
  
  ```php
  if ($ajax) {
  $attrs['x-data'] = 'blog';
}
```

```html
<div
<?php echo attrsEncode($attrs); ?>>
</div>
```
